### ISSUE TITLE
Review warning messages for precision mismatch in schema validation

### ISSUE THREAD

**Issue by helensilva14** [GoogleCloudPlatform/professional-services-data-validator#949]
It seems that this warning below is longer needed, we need to review it and remove or update it.

```
data-validation validate schema -sc oracle21 -tc bq -tbls pso_data_validator.dvt_core_types --allow-list-file=professional-services-data-validator/samples/allow_list/oracle_to_bigquery.yaml
08/21/2023 01:43:24 PM-WARNING: Source and target data type has precision mismatch: decimal(20,0) - decimal(38, 9)
08/21/2023 01:43:24 PM-WARNING: Source and target data type has precision mismatch: decimal(38,0) - decimal(76, 38)
08/21/2023 01:43:24 PM-WARNING: Source and target data type has precision mismatch: decimal(10,2) - decimal(38, 9)
```

Implemented on: https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/data_validation/schema_validation.py#L189

**Comment by nehanene15** [comment]
We log this warning when the schema of the same data type is more relaxed and the data still fits. We consider it a 'success' in schema validation, but log a warning since it's not an exact match.
See [this comment](https://github.com/GoogleCloudPlatform/professional-services-data-validator/issues/606#issuecomment-1277168783) for more details.

**Comment by nj1973** [comment]
I agree with the linked comment, but I don't think the code reflects the comment. If I read the code correctly we do this:

1. If source = target exactly then success
2. If source/target match an allow list pair then
    2.1. If target precision lower then fail
    2.2 If target precision higher then success with warning
    2.3 If precision equal then success
3. If not an exact match and not an allow list pair then fail

I think the original comment was asking for:

1. If source = target exactly then success
2. If source/target match an allow list pair then success
3. If not an exact match and not an allow list pair then
    3.1. If base type is same and target precision lower then fail
    3.2 If base type is same and target precision higher then success with warning
    3.3 Else fail

I summary it looks to me that the precision checking is in the wrong place. Hopefully I misread it.

**Comment by nehanene15** [comment]
@nj1973 Ah you're right - we are checking in the wrong place. Thanks for the explanation!
